mergify_engine/rules/filter.py

```diff
@@ -204,51 +211,76 @@ def build_evaluator(
     ) -> typing.Callable[[GetAttrObjectT], typing.Awaitable[bool]]:
         if len(tree) != 1:
             raise ParseError(tree)
+
         operator_name, nodes = list(tree.items())[0]
         try:
-            unary_op = self.unary_operators[operator_name]
+            multiple_op = self.multiple_operators[operator_name]
         except KeyError:
             try:
-                binary_op, iterable_op, compile_fn = self.binary_operators[
-                    operator_name
-                ]
+                unary_operator = self.unary_operators[operator_name]
             except KeyError:
-                raise UnknownOperator(operator_name)
-
-            nodes = typing.cast(TreeBinaryLeafT, nodes)
-            if len(nodes) != 2:
-                raise InvalidArguments(nodes)
-
-            try:
-                attribute_name, reference_value = (nodes[0], compile_fn(nodes[1]))
-            except Exception as e:
-                raise InvalidArguments(str(e))
+                try:
+                    binary_operator = self.binary_operators[operator_name]
+                except KeyError:
+                    raise UnknownOperator(operator_name)
+                nodes = typing.cast(TreeBinaryLeafT, nodes)
+                return self._handle_binary_op(binary_operator, nodes)
+            nodes = typing.cast(TreeT, nodes)
+            return self._handle_unary_op(unary_operator, nodes)
+        nodes = typing.cast(typing.List[TreeT], nodes)
```

Comment:
I feel that there's a check missing here (and probably a test too).
This is casting blindly, but nodes could be something else than a `list` and then the code would fail.

Reply:
I miss this one should be Iterator